src/cmd/compile/internal/walk/compare.go

```diff
@@ -15,6 +16,10 @@ import (
 	"cmd/compile/internal/types"
 )
 
+func fakePC() ir.Node {
+	return ir.NewInt(int64(rand.Uint32()))
```

Comment:
This will make Go compilation non-deterministic, which could very well be a blocker for getting this merged. Could we replace this with the hash of an incrementing counter seeded with e.g. the package name? Maybe ir nodes even already have such counters.

Reply:
I don't think that IR nodes have such counters. I changed the implementation to use a hash that takes into account the line number, column number, file index, and package path.